routers/common/repo.go

```diff
@@ -21,7 +22,57 @@ import (
 	"code.gitea.io/gitea/modules/util"
 )
 
-// ServeBlob download a git.Blob
+func setCommonHeaders(ctx *context.Context, name string, data interface{}) error {
+	// Google Chrome dislike commas in filenames, so let's change it to a space
+	name = strings.ReplaceAll(name, ",", " ")
+
+	ctx.Resp.Header().Set("Cache-Control", "public, max-age=300")
+
+	st, err := typesniffer.DetectContentTypeExtFirst(name, data)
+	if nil != err {
+		return err
+	}
+
+	// reset the offset to the start of served file
+	if seeker, ok := data.(io.ReadSeeker); ok {
+		_, _ = seeker.Seek(0, io.SeekStart)
+	}
+
+	if st.IsText() || ctx.FormBool("render") {
+		var cs string
+		var err error
+		if reader, ok := data.(io.ReadSeeker); ok {
+			cs, err = charset.DetectEncodingFromReader(reader)
```

Comment:
do we need a seeker reader explicit for `http.ServeContent` - else change the detection to read in a buffer and use multi reader would also be a good option